mne/io/meas_info.py

```diff
@@ -1195,7 +1195,8 @@ def _is_equal_dict(dicts):
     is_equal = []
     for d in tests:
         k0, v0 = d[0]
-        if isinstance(v0, list) and isinstance(v0[0], dict):
+        if (isinstance(v0, (list, np.ndarray)) and len(v0) > 0 and
+                isinstance(v0[0], dict)):
             for k, v in d:
```

Comment:
this isn't right, iterating over dicts iterates over the keys. Either do `for k in d:` and get the value yourself, or do `for k, v in d.items()`